### PR TITLE
Handle user interest creation.

### PR DESCRIPTION
**Pull request by shovan777**
There was a bug in user interest creation. Please remember id doesn't create relation, instance of entity is required for relation creation.

Get user interests.
Refactor code.

### CODE REVIEW

File: src/news/news.service.ts [hunk 1]
```diff
@@ -519,15 +519,53 @@ export class UserInterestsService {
...
+      interestsInputData.newsTags = await Promise.all(newsTags);
+    }
+    this.userInterestsRepository.save({
+      ...interestsInputData,
+    });
+    return interestsInputData;
+  }
+
+  async findAll(user: number): Promise<UserInterests[]> {
+    console.log(user);
```
Comment (Grunze1s) [inline src/news/news.service.ts:559]: console log remove

File: src/news/news.service.ts [hunk 2]
```diff
@@ -519,15 +519,53 @@ export class UserInterestsService {
...
+    const interestsNCount = await this.userInterestsRepository.findAndCount({
+      where: {
+        userId: user,
+        newsCategories: Not(IsNull()),
+        newsTags: Not(IsNull()),
+      },
+      relations: { newsTags: true, newsCategories: true },
     });
+    console.log(interestsNCount);
```
Comment (Grunze1s) [inline src/news/news.service.ts:568]: console.log remove

File: src/news/news.service.ts [hunk 3]
```diff
@@ -519,15 +519,53 @@ export class UserInterestsService {
...
+    return interestsNCount[0];
```
Comment (Grunze1s) [inline src/news/news.service.ts:569]: why there is response of count only ? front end would take only count if needed which in future will not break if we need to send data from same endpoint.

Comment (shovan777) [reply src/news/news.service.ts]: Done in https://github.com/atharvatechnology/ronb-node/pull/131/commits/14ecb50409f4488eaeb55e87e1670f5195e9385e

**Review by Grunze1s — APPROVED** [review]

**Pull request merged**